hack/observability/kube-state-metrics/crd-config.yaml

```diff
@@ -267,21 +283,43 @@ spec:
             failure_domain:
             - spec
             - failureDomain
-            internal_ip:
-            - status
-            - addresses
-            - "[type=InternalIP]"
-            - address
             provider_id:
             - spec
             - providerID
             version:
             - spec
             - version
-            containerRuntimeVersion:
+        type: Info
+    - name: status_nodeinfo
+      help: Node information about a machine.
+      each:
+        info:
+          labelsFromPath:
+            container_runtime_version:
             - status
             - nodeInfo
             - containerRuntimeVersion
+            internal_ip:
```

Comment:
I would probably recommend to move this into its own metric if there can be multiple internal IPs(?)
That would avoid to repeat the other fields for every IP the node has (and maybe in the future you can expose a IPv4 / IPv6 type field as well).

Reply:
@chrischdi WDYT about a dedicated `_node_ip` metric? 

With the conclusion from the ksm-slack thread in mind i would suggest to move [all the information](https://github.com/kubernetes-sigs/cluster-api/pull/7886/files#diff-06333208c85cfaff4850688d55c2b256e3210841d4e1c2cdd06bfcad30ba4587R293-R322) from the `capi_machine_status_nodeinfo` back into the already existing `capi_machine_info` metric and not introduce a new `capi_machine_node_ip` metric

Reply:
I'm +1 for adding a new metric for ips and removing `internal_ip` instead.

I'd vote for `capi_machine_ips` (which adapts from `kube_pod_ips`).

Reply:
i would go for `capi_machine_addresses` and set the five known types as label there.

Reply:
addresses matches way better because it is not always an ip :-)